### PR TITLE
feat: Code-split for admin setting for toggling appsmith watermark

### PR DESCRIPTION
**Pull request by jsartisan**
Code-split for adding a checkbox for toggling the appsmith watermark in admin settings

This PR is related to #14932

image

### CODE REVIEW

File: app/client/src/pages/AppViewer/PageMenu.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { getSelectedAppTheme } from "selectors/appThemingSelectors";
 import BrandingBadge from "./BrandingBadgeMobile";
 import { getAppViewHeaderHeight } from "selectors/appViewSelectors";
 import { useOnClickOutside } from "utils/hooks/useOnClickOutside";
-import { getShowBrandingBadge } from "@appsmith/selectors/workspaceSelectors";
+import { getAppsmithConfigs } from "ce/configs";
```
Comment (ankitakinger) [inline app/client/src/pages/AppViewer/PageMenu.tsx:23]: the import should be: `import { getAppsmithConfigs } from "**@appsmith**/configs";`

File: app/client/src/pages/AppViewer/BrandingBadge.tsx [hunk 2]
```diff
@@ -4,15 +4,10 @@ import { ReactComponent as AppsmithLogo } from "assets/svg/appsmith-logo-no-pad.
 
 function BrandingBadge() {
   return (
-    <a
-      className="fixed items-center hidden p-1 px-2 space-x-2 bg-white border rounded-md md:flex z-2 hover:no-underline right-8 bottom-4 backdrop-blur-xl backdrop-filter"
-      href="https://appsmith.com"
-      rel="noreferrer"
-      target="_blank"
-    >
+    <span className="flex items-center p-1 px-2 space-x-2 bg-white border rounded-md w-max backdrop-blur-xl backdrop-filter">
```
Comment (ankitakinger) [inline app/client/src/pages/AppViewer/BrandingBadge.tsx:7]: any reason we are removing the anchor tag?

File: app/client/src/pages/Settings/FormGroup/group.tsx [hunk 3]
```diff
@@ -126,6 +127,16 @@ export default function Group({
                     <Toggle setting={setting} />
                   </div>
                 );
+              case SettingTypes.CHECKBOX:
+                return (
+                  <div
+                    className={setting.isHidden ? "hide" : ""}
+                    data-testid="admin-settings-group-toggle"
```
Comment (ankitakinger) [inline app/client/src/pages/Settings/FormGroup/group.tsx:134]: data-testid needs to be updated

File: app/client/src/pages/Settings/FormGroup/Checkbox.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+const CheckboxWrapper = styled.div`
+  display: grid;
+  margin-bottom: 8px;
+  grid-template-columns: auto 1fr auto;
+  align-items: center;
+  gap: 16px;
+`;
+
+const StyledAuthButton = styled(Button)`
```
Comment (ankitakinger) [inline app/client/src/pages/Settings/FormGroup/Checkbox.tsx:26]: This is the StyledUpgradeButton if I'm not wrong?

**Review by ankitakinger — CHANGES_REQUESTED** [review]
Could you please add a loom video showing these changes on Admin settings page?

**Comment by jsartisan** [comment]
@ankitakinger This PR will add a UI to toggle watermark in super admin settings page. But for CE, it will be disabled
image

**Review by ankitakinger — DISMISSED** [review]

**Review by ankitakinger — APPROVED** [review]

**Comment by ankitakinger** [comment]
@jsartisan can we add unit and cypress tests for the new files/changes?

**Comment by jsartisan** [comment]
@ankitakinger Added the cypress tests for the setting. Checking if the setting has a upgrade button. 
https://github.com/appsmithorg/appsmith/pull/15036/commits/3dd11cbe863e93bac3c95ec4ad313f9d8ea0a497

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**